src/Base/Contracts/BaseContract.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace IndieHD\Velkart\Base\Contracts;
+
+use Illuminate\Database\Eloquent\Collection;
+
+interface BaseContract
```

Comment:
Should this interface be named `BaseRepositoryContract`, instead of just `BaseContract`? After all, it seems specific to repositories.